src/Adapter/MSTest.TestAdapter/Discovery/TypeValidator.cs

```diff
@@ -106,7 +106,7 @@ internal static bool HasCorrectTestContextSignature(Type type)
     {
         DebugEx.Assert(type != null, "HasCorrectTestContextSignature type is null");
 
-        IEnumerable<PropertyInfo> propertyInfoEnumerable = PlatformServiceProvider.Instance.ReflectionOperations.GetDeclaredProperties(type);
+        PropertyInfo[] propertyInfoEnumerable = PlatformServiceProvider.Instance.ReflectionOperations.GetDeclaredProperties(type);
```

Comment:
Now that the type is array, this should avoid a boxing allocation for the enumerator.